src/GitHub.VisualStudio/Services/VSGitExtFactory.cs

```diff
@@ -0,0 +1,49 @@
+﻿extern alias TF14;
+extern alias TF15;
+
+using System;
+using System.Threading.Tasks;
+using System.ComponentModel.Composition;
+using GitHub.Logging;
+using Serilog;
+using EnvDTE;
+using Microsoft.VisualStudio.Shell;
+using VSGitExt14 = TF14.GitHub.VisualStudio.Base.VSGitExt;
+using VSGitExt15 = TF15.GitHub.VisualStudio.Base.VSGitExt;
+
+namespace GitHub.Services
+{
+    [PartCreationPolicy(CreationPolicy.Shared)]
+    public class VSGitExtFactory
+    {
+        static readonly ILogger log = LogManager.ForContext<VSGitExtFactory>();
+
+        readonly IGitHubServiceProvider serviceProvider;
+
+        [ImportingConstructor]
+        public VSGitExtFactory(IGitHubServiceProvider serviceProvider)
+        {
+            this.serviceProvider = serviceProvider;
+        }
+
+        [Export(typeof(IVSGitExt))]
+        public IVSGitExt VSGitExt => serviceProvider.GetService<IVSGitExt>();
+
+        public async static Task<IVSGitExt> Create(IAsyncServiceProvider sp)
+        {
+            var dte = await sp.GetServiceAsync(typeof(DTE)) as DTE;
```

Comment:
`ApplicationInfo.GetVersion` will give you the running VS version as a proper type, so you don't have to grab a separate service for that here